### DISCUSSION TITLE
Is there a way to avoid META while editing Fontawesome icon as links?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#858] [Q&A] [answered]
Hi,

What's the best way to edit the `href` attribute when using Font Awesome icons like 

`<i class="fa-brands fa-instagram"></i>`

which is inside a link `<a>` element

I saw mavo's demo [here](https://mavo.io/demos/homepage/), but i would like to know if there's a way to use it in a much more simple way just adding <i></i>

**Comment by DmitrySharabin** [comment]
Could you, please, give me a bit more context so that I can understand the problem you are trying to solve?

**Comment by GalinhaLX** [comment]
Hi

I've made this [codepen](https://codepen.io/galinhalx/pen/KKQyWeE)

**Comment by GalinhaLX** [comment, marked as answer]
Ok, so I figure it out:

@LeaVerou 's workaround to avoid a element click just works on ...`a` elements

When using @DmitrySharabin workaround, I can use a specific class, and apply it to icon (great)

New codepen, [here](https://codepen.io/galinhalx/pen/oNEoWwE) (see solution 2)


Unfortunately, I don't know javascript enought: 

Would you recommend to adapt @LeaVerou workaround to have ability to add other elements, or classes?
Or is @DmitrySharabin workaround doing all good already?


By the way, I'm looking to have the editing effect just like in this demo, BUT using `<i> `elements from FontAwesome

<img width="489" alt="image" src="https://user-images.githubusercontent.com/89095737/170296691-acbddf7a-5153-4a44-a1d7-bb690e5585e0.png">

**Reply by DmitrySharabin** [reply]
Do you mean something like this? https://codepen.io/dmitrysharabin/pen/bGLYoQN

A couple of notes:
- To be able to edit `href` of an` <a>` element, it needs to be a _property_ (and it has to have a value so that the contents of the `<a>` element is to be shown in read mode)
- You can use expressions as a part of a complex class name (e.g., in the pen above, I used the [between](https://mavo.io/docs/functions/#between) function to get the name of the social network from the URL).

Hope it helps!

**Reply by GalinhaLX** [reply]
> Do you mean something like this? https://codepen.io/dmitrysharabin/pen/bGLYoQN

Yes ! 
...Except 

- it doesn't work my side (both Firefox, chrome) -> Opens popup, then follows the link next second
- ...but it works on Safari :)

> 
> A couple of notes:
> 
>     * To be able to edit `href` of an` <a>` element, it needs to be a _property_ (and it has to have a value so that the contents of the `<a>` element is to be shown in read mode)
> 

Do you mean a mv-default value? Or a value in the inner html?
In your codepen, does the `<i></i>` element counts as a value for the parent element where you set the property? 
(Because I don't see any value in the codepen)


>     * You can use expressions as a part of a complex class name (e.g., in the pen above, I used the [between](https://mavo.io/docs/functions/#between) function to get the name of the social network from the URL).

WOW! That's great !
[Documentation](https://mavo.io/docs/functions/#between) could be extended with links to your codepen
Because without examples, it's quite difficult for someone with an entry level (like me) to understand how it should work

> 
> 
> Hope it helps!

On the way to :)

PS: Nice trick to use profile to grab the social icon from RemixIcons ! :)

**Reply by GalinhaLX** [reply]
It seems  @DmitrySharabin 's code works better

[Codepen](https://codepen.io/galinhalx/pen/bGLYvEp)

If I take your codepen, and apply your workaround to edit <a>, it now works for **Firefox, Chrome & Safari**

Many thanks !

**Reply by DmitrySharabin** [reply]
> Do you mean a mv-default value? Or a value in the inner html?

Sorry for my poor explanation and let me fix it. :) If an `<a>` element is a _property_, its value is stored (by default) in the `href` attribute. So, the `<a>` property has a value if: 1) it has the `href` attribute and 2) the value of this attribute is not an empty string.

For example, if we have `<a property="link" href="https://mavo.io/">Foo</a>`, the `link` property has the value of `https://mavo.io/`.

In the following two cases, the `foo` and `bar` properties have no values (and the user won't see the **Foo** and **Bar** text in read mode):
1. `<a property="foo">Foo</a>` (no `href` attribute)
2. `<a property="bar" href="">Bar</a>` (the value of `href` is an empty string).

And btw, `mv-default` would work too.

>In your codepen, does the` <i></i>` element counts as a value for the parent element where you set the property?

Absolutely. Anything between `<a>` and `</a>` is the content of the `<a>` element.

**Reply by DmitrySharabin** [reply]
>it doesn't work my side (both Firefox, chrome) -> Opens popup, then follows the link next second

You're right! I would suggest using this code snippet as a workaround (I modified Lea's workaround, so it reflects our case):
```js
document.addEventListener("click", (evt) => {
	if (evt.target.closest('[mv-mode="edit"] a')) {
		evt.preventDefault();
	}
});
```

I already fixed my [pen](https://codepen.io/dmitrysharabin/pen/bGLYoQN).

**Reply by GalinhaLX** [reply]
Hi @DmitrySharabin 

Thank you so much for these explanations !

About your code:
- It perfectly works under Chrome & Firefox (edit , and follows link)
- On **Safari**: edit mode (OK), but in 'read' mode **doesn't work to follow the link** :(

**Reply by DmitrySharabin** [reply]
It's because of the way CodePen works (using frames) and some Safari restrictions.

<img width="854" alt="Screen Shot 2022-05-26 at 16 20 39" src="https://user-images.githubusercontent.com/9166277/170507089-1abc38fa-ae95-49e7-89d5-b3dd11da7c2a.png">

If you open the pen in Debug mode, it'll work.

<img width="524" alt="Screen Shot 2022-05-26 at 16 24 02" src="https://user-images.githubusercontent.com/9166277/170507801-8d43f91a-1687-494d-807e-f30a0863e9a0.png">

**Reply by GalinhaLX** [reply]
oh, thanks for information !

and congratulations for the new code :) safe to use it so ;)
